src/AutoRest.CSharp/Common/Output/Models/ParameterInvocation.cs

```diff
@@ -1,22 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
-using AutoRest.CSharp.Generation.Types;
-using AutoRest.CSharp.Generation.Writers;
-using AutoRest.CSharp.Output.Models.Shared;
-
-namespace AutoRest.CSharp.Common.Output.Models
-{
-    internal record ParameterInvocation : Parameter
-    {
-        public ParameterInvocation(Parameter parameter, CodeWriterDelegate? Invocation = default) : this(parameter.Name, parameter.Description, parameter.Type, parameter.DefaultValue, parameter.ValidateNotNull, IsApiVersionParameter: parameter.IsApiVersionParameter, IsResourceIdentifier: parameter.IsResourceIdentifier, SkipUrlEncoding: parameter.SkipUrlEncoding, RequestLocation: parameter.RequestLocation, Invocation: Invocation) { }
```

Comment:
This type has problems in several places at once:
1. Semantically, `Parameter` type is used to represent parameter in method signature. It has some properties that are used in parameter validation logic, but it is still part of the declaration. `Invocation` property has been used only in initializers or constructor calls, making the purpose of the `ParameterInvocation` type completely different, yet dragging all the properties from the base type, even though they are useless
2. `ParameterInvocation(Parameter parameter, CodeWriterDelegate? Invocation = default)` constructor's first argument is used only to initialize unused fields, hence it was also useless. Yet, when `ParameterInvocation` was created, it was always set to some value.
3. `CodeWriterDelegate` is a type for special cases and should be used only as a last resort. In addition to general problems with callbacks (very complicated code analysis for those who are new to the codebase, inverted stack trace during debugging, accidentally captured variables and unexpected caching, etc.), there is one specific to `CodeWriter` - we have no way to limit the scope of the callback call (see examples below).